mev_inspect/compound_liquidations.py

```diff
@@ -0,0 +1,126 @@
+            c_token_mapping[
+                c_token.lower()
+            ] = underlying_token.lower()  # make k:v lowercase for consistancy
+    return c_token_mapping
+
+def get_underlying_ctoken_exchange_rate(
+    c_token_address: str, block_number: int, w3: Web3, comp_markets: Dict[str, str]
+) -> float:
+    comp_v2_ctoken_abi = get_raw_abi("CToken", Protocol.compound_v2)
+    ctoken_instance = w3.eth.contract(address=Web3.toChecksumAddress(c_token_address), abi=comp_v2_ctoken_abi)
+    raw_exchange_rate = ctoken_instance.functions.exchangeRateCurrent().call(
+        block_identifier=block_number
+    )
+    # format based on decimals in ctoken and the underlying token
+    # see "Interpreting Exchange Rates" https://compound.finance/docs#protocol-math
+    underlying_token_address = comp_markets[c_token_address.lower()]
+    decimals_in_underlying = get_erc20_token_decimals(
+        underlying_token_address,
+        w3
+    )
+    decimals_in_ctoken = get_erc20_token_decimals(c_token_address, w3)
+    return raw_exchange_rate / (
+        10 ** (18 + decimals_in_underlying - decimals_in_ctoken)
+    )
+
+def get_erc20_token_decimals(token_address: str, w3: Web3) -> int:
+    token_abi = get_raw_abi("ERC20", None)
+    token_instance = w3.eth.contract(address=Web3.toChecksumAddress(token_address), abi=token_abi)
+    decimals = token_instance.functions.decimals().call()
+    return decimals
+
+def get_compound_liquidations(
+    traces: List[ClassifiedTrace],
+    comp_markets: Dict[str,str],
+    w3: Web3,
+) -> List[Liquidation]:
+
+    """Inspect list of classified traces and identify liquidation"""
+    liquidations: List[Liquidation] = []
+
+    for trace in traces:
+
+        if (
+            trace.classification == Classification.liquidate
+            and trace.protocol == Protocol.compound_v2
+        ):
+            # First, we look for cEther liquidations (position paid back via tx.value)
+            child_traces = get_child_traces(trace.transaction_hash, trace.trace_address, traces)
+            seize_trace = _get_seize_call(child_traces)
+            c_token_collateral = trace.inputs["cTokenCollateral"]
+            underlying_exchange_rate = get_underlying_ctoken_exchange_rate(c_token_collateral, trace.block_number, w3, comp_markets)
+            if trace.abi_name =='CEther':
+                liquidations.append(
+                    Liquidation(
+                        liquidated_user=trace.inputs["borrower"],
+                        collateral_token_address=WETH_ADDRESS, # WETH since all cEther liquidations provide Ether
+                        debt_token_address=comp_markets[c_token_collateral],
+                        liquidator_user=seize_trace.inputs["liquidator"],
+                        debt_purchase_amount=trace.value,
+                        protocol=Protocol.compound_v2,
+                        received_amount=seize_trace.inputs["seizeTokens"] * underlying_exchange_rate,
```

Comment:
the intention for `received_amount` was to be denominated in the received token

I know for aave @gheise is working on just looking at the underlying transfer traces to get this value

if that's possible here it would save us from having to look up exchange rates